### DISCUSSION TITLE
How to execute a .cmd file using Visual Studio Developer Command prompt in Github actions?

### DISCUSSION THREAD

**Discussion by AnkitKaushalTR** [community/community#26838] [Actions]
<p>We are having a .cmd script file which i want to run using visual studio developer command prompt inside the github actions, is it possible ?</p>
<p>If yes, then can you guys help with the syntax that i have to provide in the workflow yaml file?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/ankitkaushaltr">@ankitkaushaltr</a> ,</p>
<p>Checked the VS Develper Command prompt <a href="https://docs.microsoft.com/en-us/dotnet/framework/tools/developer-command-prompt-for-vs#manually-locate-the-files-on-your-machine" rel="nofollow noopener">execution path</a>, please try with below code:</p>
<pre><code>jobs:
  job1:
    runs-on: [Windows-latest]
    steps:
      - name: checkout
        uses: actions/checkout@v2
      - name: shellcheck
        shell: cmd
        run: |
          call "C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Visual Studio 2019\Visual Studio Tools\Developer Command Prompt for VS 2019.lnk"
          call ${{ github.workspace}}\test.cmd
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bce6f96f21539aaeaedb8e77ec4b1237d52cd36f.png" data-download-href="/uploads/short-url/qX6CJob1QaC9q4DGJDZN6eMGlpt.png?dl=1" title="batchcmd.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181100245-70b5344f-30bb-49a4-9edd-86477ad1a3b7.png" alt="batchcmd.png" data-base62-sha1="qX6CJob1QaC9q4DGJDZN6eMGlpt" width="690" height="133" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bce6f96f21539aaeaedb8e77ec4b1237d52cd36f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">batchcmd.png</span><span class="informations">1105×213 13.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks</p>

**Comment by jwillemsen** [comment]
<p>You can also use the msvc-dev-cmd action, see <a href="https://github.com/marketplace/actions/enable-developer-command-prompt" rel="noopener nofollow ugc">https://github.com/marketplace/actions/enable-developer-command-prompt</a>, that works with multiple visual studio versions</p>
